grpc/src/main/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerBuilder.java

```diff
@@ -0,0 +1,243 @@
+     * @param jsonMarshaller The custom JSON marshaller to use
+     */
+    public UnframedGrpcErrorHandlerBuilder jsonMarshaller(MessageMarshaller jsonMarshaller) {
+        requireNonNull(jsonMarshaller, "jsonMarshaller");
+        if (marshalledMessages != null || marshalledMessageTypes != null) {
+            throw new IllegalArgumentException(
+                    "Cannot set a custom JSON marshaller because one or more Message or MessageType instances" +
+                    " have already been registered. To set a custom marshaller, ensure that no Message or" +
+                    " MessageType registrations have been made before calling this method.");
+        }
+        this.jsonMarshaller = jsonMarshaller;
+        return this;
+    }
+
+    /**
+     * Specifies the status mapping function to be used by the error handler.
+     *
+     * <p>This function determines how gRPC statuses are mapped to HTTP statuses
+     * in the error response.
+     *
+     * @param statusMappingFunction The status mapping function
+     */
+    public UnframedGrpcErrorHandlerBuilder statusMappingFunction(
+            UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = requireNonNull(statusMappingFunction, "statusMappingFunction");
+        return this;
+    }
+
+    /**
+     * Specifies the response types that the error handler will support.
+     *
+     * <p>This method allows specifying one or more response types (e.g., JSON, PLAINTEXT)
+     * that the error handler can produce. If nothing is specified or multiple types are specified, the actual
+     * response type is determined by the response's content type.
+     *
+     * @param responseTypes The response types to support
+     */
+    public UnframedGrpcErrorHandlerBuilder responseTypes(UnframedGrpcErrorResponseType... responseTypes) {
+        requireNonNull(responseTypes, "responseTypes");
+        if (this.responseTypes == null) {
+            this.responseTypes = EnumSet.noneOf(UnframedGrpcErrorResponseType.class);
+        }
+        this.responseTypes.addAll(ImmutableSet.copyOf(responseTypes));
+        return this;
+    }
+
+    /**
+     * Registers custom messages to be marshalled by the error handler.
+     *
+     * <p>This method registers specific message instances for custom error responses.
+     * If a custom JSON marshaller has already been set, calling this method will
+     * result in an {@link IllegalArgumentException}.
+     *
+     * @param messages The message instances to register
+     */
+    public UnframedGrpcErrorHandlerBuilder registerMarshalledMessages(Message... messages) {
+        requireNonNull(messages, "messages");
+        if (jsonMarshaller != null) {
+            throw new IllegalArgumentException(
+                    "Cannot register custom messages because a custom JSON marshaller has already been set. " +
+                    "Use the custom marshaller to register custom messages.");
+        }
```

Comment:
In this case, the argument itself is not bad but the builder's state is, so we could do this:

```java
checkState(jsonMarshaller == null, "Cannot ...");
```